cloudevents/encoding_structured.go

```diff
@@ -64,13 +62,24 @@ func (structured) FromRequest(data interface{}, r *http.Request) (*EventContext,
 	rawData := raw["data"]
 	delete(raw, "data")
 
-	// TODO: support both v01 and v02 encoding
-	ec := &EventContext{}
+	var ec LoadContext
+	v := ""
+	if err := json.Unmarshal(raw["specversion"], &v); err == nil && v == V02CloudEventsVersion {
+		ec = &V02EventContext{}
+	} else if err := json.Unmarshal(raw["cloudEventsVersion"], &v); err == nil && v == V01CloudEventsVersion {
+		ec = &V01EventContext{}
+	} else {
+		// For historical reasons, this library treated unknown payloads as V01. Preserve for now.
+		ec = &V01EventContext{}
+		// TODO: change behavior to be an error:
```

Comment:
Issue#/Owner?

Reply:
I'm going to do it and/or kill off V01 altogether.

If you want, I can make that part of this PR; I was feeling some "change fatigue" last night.

Reply:
If it's just around the corner, it doesn't matter, I guess.

Reply:
I fixed this, it turned out to be 2-3 tests that needed the CloudEventsVersion field to be set.